### DISCUSSION TITLE
NFStream Only Extracting Flows of Only One host from a PCAP File

### DISCUSSION THREAD

**Discussion by maverickwh01** [nfstream/nfstream#66] [General]
I'm Testing NF Stream on Both the online interface mode as well as the offline mode using a pcap file and in both cases it is only showing  the flows corresponding the host on the streamer is deployed to and in case of the pcap file it is only showing flows of the host which was used to capture the pcap. Is there a way to extract all the flows corresponding to the hosts in the present pcap and likewise all the hosts whose packets are being monitored. 

Thanks in Advance.

**Comment by drnpkr** [comment]
Hi,

I think what you experience is active/idle timeout in action. In online mode the flow is written to a CSV only once it has expired. If you use the default config, only after 120 seconds you will see the flow written out in the CSV. In offline mode, if you stop PCAP processing prematurely, the same thing happens. The flow is cached. If the processing finishes, you will get the other flows. This is a design choice implemented in NFStream to achieve high performance.

I suggest you test NFStream using different timeout configurations. Otherwise, if you still observe only one flow, there might be another issue. I such a case, I suggest you share the PCAP file with us.

**Reply by maverickwh01** [reply]
I'll surely test with different timeouts. Meanwhile please provide me your email so i can share the pcap file with you.

Thanks

**Reply by drnpkr** [reply]
It is: apekar[at]hit[dot]bme[dot]hu
